cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -56,6 +85,73 @@ XDMFFile::XDMFFile(MPI_Comm comm, const std::string filename)
   // Flush datasets to disk at each timestep. Allows inspection of the
   // HDF5 file whilst running, at some performance cost.
   parameters.add("flush_output", false);
+
+  // Synchronize before filesystem check
+  _mpi_comm.barrier();
```

Comment:
Why a barrier?

Reply:
Since only root process saves XML file, it could happen, that root is slow and didn't save the XML file of previous write method, while other faster process continues and possibly constructs file in "r" mode but apparently can't find the XML file so prepares new empty structure. This process then has no information on where in HDF file (hdf path) the object to read could be.